client/app/components/dynamic-form/DynamicForm.jsx

```diff
@@ -57,8 +57,13 @@ class DynamicForm extends React.Component {
     super(props);
 
     const hasFilledExtraField = some(props.fields, field => {
-      const { extra, initialValue } = field;
-      return extra && (!isEmpty(initialValue) || isNumber(initialValue) || (isBoolean(initialValue) && initialValue));
+      const { extra, initialValue, placeholder } = field;
+      return (
+        extra &&
+        (!isEmpty(initialValue) ||
+          isNumber(initialValue) ||
+          (isBoolean(initialValue) && initialValue.toString() !== placeholder))
```

Comment:
Shouldn't `initialValue.toString() !== placeholder` check be applied to other field types?

Reply:
For the other fields we have the `isEmpty` check. But booleans don't have an empty state, so we need to compare with the default.